staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -929,3 +934,73 @@ func isWatchErrorRetriable(err error) bool {
 	}
 	return false
 }
+
+type initialEventsEndBookmarkTicker struct {
+	clock.Ticker
+	clock clock.Clock
+	name  string
+
+	watchStart           time.Time
+	tickInterval         time.Duration
+	lastEventObserveTime time.Time
+}
+
+func newInitialEventsEndBookmarkTicker(name string, c clock.Clock, watchStart time.Time, exitOnInitialEventsEndBookmark *bool) *initialEventsEndBookmarkTicker {
+	return newInitialEventsEndBookmarkTickerInternal(name, c, watchStart, 10*time.Second, exitOnInitialEventsEndBookmark)
+}
+
+func newInitialEventsEndBookmarkTickerInternal(name string, c clock.Clock, watchStart time.Time, tickInterval time.Duration, exitOnInitialEventsEndBookmark *bool) *initialEventsEndBookmarkTicker {
```

Comment:
for easier testing, so that we don't have to overwrite `initialEventsEndBookmarkTicker.Ticker` and `initialEventsEndBookmarkTicker.tickInterval`